src/AccessControl/SelectorAccessControl.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import {ContextUpgradeable} from "openzeppelin-contracts-upgradeable/utils/ContextUpgradeable.sol";
+import {AccessControlEnumerableUpgradeable} from
+    "openzeppelin-contracts-upgradeable/access/extensions/AccessControlEnumerableUpgradeable.sol";
+import {
+    AccessControlUpgradeable,
+    IAccessControl
+} from "openzeppelin-contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import {EVCUtil} from "ethereum-vault-connector/utils/EVCUtil.sol";
+
+/// @title SelectorAccessControl
+/// @custom:security-contact security@euler.xyz
+/// @author Euler Labs (https://www.eulerlabs.com/)
+/// @notice A utility contract with the EVC support that allows for access control based on specific selectors.
+abstract contract SelectorAccessControl is EVCUtil, AccessControlEnumerableUpgradeable {
+    /// @notice The wildcard for all selectors. A caller with this role can call any function selector.
+    bytes32 public constant WILD_CARD = bytes32(type(uint256).max);
+
+    /// @notice Constructor for SelectorAccessControl
+    /// @param evc The address of the Ethereum Vault Connector (EVC)
+    /// @param admin The address to be granted the DEFAULT_ADMIN_ROLE
+    constructor(address evc, address admin) EVCUtil(evc) {
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+        _disableInitializers();
+    }
+
+    /// @notice Initializes the contract, setting up the admin role
+    /// @param admin The address to be granted the DEFAULT_ADMIN_ROLE
+    function initialize(address admin) public initializer {
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+    }
+
+    /// @dev Grants `role` to `account`.
+    function grantRole(bytes32 role, address account)
+        public
+        virtual
+        override (AccessControlUpgradeable, IAccessControl)
+        onlyEVCAccountOwner
```

Comment:
It's too bad we have to remember to override any functions that should have `onlyEVCAccountOwner`... I wouldn't worry about it here, but in the future maybe we should have a variant method in EVCUtil for this, like `_msgSenderOnlyOwner()` ?

Reply:
It wouldn't work. All OZ contracts rely on `_msgSender()` function in their implementation. Introducing a new function like `_msgSenderOnlyOwner()` will mean that we also need to override all the functions that currently use `_msgSender()` and that we'd like to be decorated with `onlyEVCAccountOwner` to use this new `_msgSenderOnlyOwner()` function. It would get even worse.

Reply:
I believe you, but I don't understand why you'd have to override additional stuff. My idea is to change what function is invoked in our own override of `_msgSender()`, ie here in `SelectorAccessControl` contract:

```
    function _msgSender() internal view virtual override (EVCUtil, ContextUpgradeable) returns (address) {
        return EVCUtil._msgSender();
    }
```

`_msgSenderOnlyOwner()` would do the exact same thing as `EVCUtil._msgSender()` except it would revert if not owner.

Reply:
let's say that we want to build the following smart contract that inherits from two other OZ contracts:

```solidity
contract My_Contract is EVCUtil, ERC20, Ownable {

   ...

   function _msgSender() internal view virtual override (EVCUtil, Context) returns (address) {
      return EVCUtil._msgSenderOnlyOwner();
   }
}
```

The above wouldn't work because the `_msgSender()` is used both in the `ERC20` and `Ownable` while we only would like the above to be applied in the `Ownable`. This is because we only need the effect of using `onlyEVCAccountOwner` for the governance-related functions, not all off them. ERC20 shouldn't be affected

Reply:
As discussed today on the call, we'd only use `_msgSenderOnlyOwner()` for "simple" contracts like this one, that don't need sub-accounts/etc for any methods.